.gitignore

```diff
@@ -71,3 +71,6 @@ doc/Makefile
 cscope.*
 ncscope.*
 
+# dependencies
```

Comment:
I think we also have something in the python/bash build scripts that check/report on submodules. This would need updated too.
Fwiw, the python/bash scripts should probably also print out that they are being deprecated for Intel/NXP and west should be used instead as part of this PR.